crates/api_models/src/payments.rs

```diff
@@ -1224,6 +1227,9 @@ pub struct Metadata {
     pub data: pii::SecretSerdeValue,
     /// Payload coming in request as a metadata field
     pub payload: Option<pii::SecretSerdeValue>,
+
+    /// Allowed payment method types for a payment intent
+    pub allowed_payment_method_types: Option<Vec<api_enums::PaymentMethodType>>,
```

Comment:
We should be populating the response too, the metadata struct should be parsed and passed here. You can check the [payments_to_payments_response](https://github.com/juspay/hyperswitch/blob/43aaf8f735eb2b74e12ab0fe49026dd6767e546e/crates/router/src/core/payments/transformers.rs#L233) function, and add this field there.